Neolution.Extensions.Identity/JwtSignInManager.cs

```diff
@@ -0,0 +1,116 @@
+﻿namespace Neolution.Extensions.Identity
+{
+    using System.Security.Claims;
+    using Google.Apis.Auth;
+    using Microsoft.AspNetCore.Identity;
+    using Neolution.Extensions.Identity.Abstractions;
+    using JsonWebToken = Neolution.Extensions.Identity.Abstractions.JsonWebToken;
+
+    /// <inheritdoc/>
+    public sealed class JwtSignInManager<TUser> : IJwtSignInManager<TUser>
+        where TUser : IdentityUser<Guid>
+    {
+        /// <summary>
+        /// The user manager
+        /// </summary>
+        private readonly IUserManager<TUser> userManager;
+
+        /// <summary>
+        /// The sign in manager
+        /// </summary>
+        private readonly ISignInManager<TUser> signInManager;
+
+        /// <summary>
+        /// The JWT generator
+        /// </summary>
+        private readonly IJwtGenerator<TUser> jwtGenerator;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="JwtSignInManager{TUser}"/> class.
+        /// </summary>
+        /// <param name="userManager">The user manager.</param>
+        /// <param name="signInManager">The sign in manager.</param>
+        /// <param name="jwtGenerator">The JWT generator.</param>
+        public JwtSignInManager(IUserManager<TUser> userManager, ISignInManager<TUser> signInManager, IJwtGenerator<TUser> jwtGenerator)
+        {
+            this.userManager = userManager;
+            this.signInManager = signInManager;
+            this.jwtGenerator = jwtGenerator;
+        }
+
+        /// <inheritdoc/>
+        public async Task<JsonWebToken?> SignInWithGoogleAsync(string token, string clientId)
+        {
+            var validationSettings = new GoogleJsonWebSignature.ValidationSettings
+            {
+                Audience = new List<string> { clientId },
+            };
+
+            var payload = await GoogleJsonWebSignature.ValidateAsync(token, validationSettings); // Will throw an exception if validation fails.
+
+            // TODO: Think about different discovery options for external users
+            var user = await this.userManager.FindByEmailAsync(payload.Email);
+            if (user is null)
+            {
+                return null;
+            }
+
+            var canSignInResult = await this.signInManager.PreSignInCheckAsync(user);
+            if (canSignInResult != null)
```

Comment:
This check should check if the canSignInResult is null or ifs not succeeded.

Reply:
Fixed